specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2021-07-01-preview/loadtestservice.json

```
@@ -0,0 +1,2639 @@
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "description": "[ReadOnly] Fully qualified resource Id e.g subscriptions/{subId}/resourceGroups/{rg}/providers/Microsoft.LoadTestService/loadtests/{resName}",
+          "type": "string"
+        },
+        "resourceName": {
+          "description": "Azure resource name",
+          "type": "string"
+        },
+        "resourceType": {
+          "description": "Azure resource type",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "[ReadOnly] Azure resource display name",
+          "type": "string"
+        },
+        "resourceGroup": {
+          "description": "[ReadOnly] Resource group name of Azure resource",
+          "type": "string"
+        },
+        "subscriptionId": {
+          "description": "[ReadOnly] Subscription Id of Azure resource",
+          "type": "string"
+        },
+        "kind": {
+          "description": "Kind of Azure resource type",
+          "type": "string"
+        }
+      }
+    },
+    "AppComponentsMap": {
+      "description": "App components model",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "description": "[ReadOnly] Load test resource id",
+          "type": "string"
+        },
+        "testId": {
+          "description": "[Required if testRunId is not given] Load test unique identifier",
+          "type": "string"
+        },
+        "testRunId": {
+          "description": "[Required if testId is not given] Load test run unique identifier",
+          "type": "string"
+        },
+        "name": {
+          "description": "[ReadOnly] AppComponent name",
+          "type": "string"
+        },
+        "value": {
+          "description": "AppComponents Map { resource id (Fully qualified resource Id e.g subscriptions/{subId}/resourceGroups/{rg}/providers/Microsoft.LoadTestService/loadtests/{resName}) : resource object } ",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/AppComponent"
+          }
+        }
+      }
+    },
+    "ErrorResponseBody": {
+      "description": "The error response object.",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "The error code.",
+          "type": "string"
+        },
+        "message": {
+          "description": "The error message.",
+          "type": "string"
+        },
+        "target": {
+          "description": "The error target.",
+          "type": "string"
+        },
+        "details": {
+          "description": "Additional details and inner errors.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ErrorResponseBody"
+          }
+        }
+      }
```

The `code` and `message` properties should be `required` (always included):
```suggestion
      },
      "required": [
        "code",
        "message"
      ]
```

---

It will be addressed in the next API revision